### ISSUE TITLE
BUG: Missing spaces/newlines in saved rotation matrix/selection box?

### ISSUE THREAD

**Issue by pauldmccarthy** [neuroanatomy/reorient#3]
## Thank you for your feedback.

Hi @r03ert0 and @katjaq, this is a really nice tool! I've been playing around with it as part of openjournals/joss-reviews#2670, and have found a couple of strange behaviours - I will open one or two more issues after this one (sorry for the deluge!).

I am not 100% sure if they are bugs, or browser-specific quirks, but I thought I'd bring them to your attention just in case. 

### Current behaviour

When using the _Save Rotation Matrix_ and _Save Selection Box_ options, I end up with text files which appear to be missing delimiters (probably new lines). An example rotation matrix which I get is ([also attached](https://github.com/neuroanatomy/reorient/files/5322113/reorient.txt)):

```
-0.8210228572672786 0.5441567480768857 -0.1726699202729373 35.392545939637480.42533908247618385 0.7847880104434116 0.45077094358730546 -155.19998034522737-0.38079933398443844 -0.2966499826169898 0.8757800266335992 -22.7527181620778830 0 0 1
```

Note that there are no newlines in the file - I think that the following is what should be produced:

```
-0.8210228572672786 0.5441567480768857 -0.1726699202729373 35.39254593963748
0.42533908247618385 0.7847880104434116 0.45077094358730546 -155.19998034522737
-0.38079933398443844 -0.2966499826169898 0.8757800266335992 -22.752718162077883
0 0 0 1
```

Similar behaviour occurs when using the _Save Selection Box_ option - I end up with a `selection.txt` file which looks like this ([also attached](https://github.com/neuroanatomy/reorient/files/5322112/selection.txt)):

```
-30 -96 -7930 30 67
```

But I think what is probably correct is this:

```
-30 -96 -79
30 30 67
```
### Version information (for bug reports)

* **Browser + version**: Firefox 80.0.1
* **Your OS + version**: Ubuntu 18.04

**Comment by r03ert0** [comment]
Thank you @pauldmccarthy !
How strange. We hadn't checked with Firefox, only Chrome (and our tests run on Chromium).
Here's the `saveMatrix` function, which does have a `\n` for line breaks:

```js
function saveMatrix() {
  const {mv} = globals;
  const a = document.createElement('a');
  const m = mv.mri.MatrixVox2Mm;
  const str = m.map((o) => o.join(' ')).join('\n');
  a.href = 'data:text/csv;charset=utf-8,' + str;
  const name = prompt("Save Voxel To World Matrix (the inverse of the one displayed) As...", "reorient.mat");
  if(name !== null) {
    a.download=name;
    document.body.appendChild(a);
    a.click();
  }
}
```

We'll check in Firefox (also Safari, but maybe not Explorer :p [maybe we can add a warning])

**Comment by pauldmccarthy** [comment]
I just tested on google chrome (still using Ubuntu 18.04), and indeed, the files save correctly :) So it seems to be firefox-specific. I'm not a Javascript expert, but your code looks pretty straightforward, so I can't imagine what is going on ...

**Comment by r03ert0** [comment]
Fixed in commit e2e6eb9
The plain text string goes into a URL. Chrome is ok with `\n`, but Firefox wants it properly translated to `%0A` : )
(tested in Firefox 80.0.1)
